akka-actor/src/main/scala/akka/util/MessageBuffer.scala

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.util
+
+import akka.actor.ActorRef
+
+/**
+ * A single producer single consumer mutable message buffer
+ */
+final class MessageBuffer private (
+  private var _head: MessageBuffer.Node,
+  private var _tail: MessageBuffer.Node) {
```

Comment:
I'm not sure what benefits that would bring, since you would have to allocate an object to hold the two values just to put it into the `ArrayDeque`. Do you have special use case in mind?